app/main.py

```diff
@@ -1,8 +1,45 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self, comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self, distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = average_rating
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]) -> float:
+        total_income = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                wash_price = self.calculate_washing_price(car)
+                total_income += wash_price
+                self.wash_single_car(car)
+        return round(total_income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        if self.clean_power > car.clean_mark:
+            different = (self.clean_power - car.clean_mark)
+            pre_price = car.comfort_class * different * self.average_rating
+            price = (pre_price / self.distance_from_city_center)
+            return round(price, 1)
+        return 0.0
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, new_rating: float) -> None:
+        total_rating = self.average_rating * self.count_of_ratings
+        self.count_of_ratings += 1
+        new_rating = (total_rating + new_rating) / self.count_of_ratings
```

Comment:
The calculation of the new average rating is quite complex and it would be easier to understand if it was split into several steps. You can use additional variables to store intermediate results or break the calculation after binary operations. For example, you could calculate the 'total_rating' and 'new_rating' in separate lines and then use these values to calculate the 'average_rating'. This would make your code more readable and maintainable. (Checklist: Code Style 1)